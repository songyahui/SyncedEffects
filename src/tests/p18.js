"use hopscript"

var hh = require("hiphop");

var prg = <hh.module S1_and_S2 S1_and_not_S2 not_S1_and_S2 not_S1_and_not_S2>
  <hh.loop>
    <hh.trap T1>
      <hh.local S1>
	<hh.parallel>
	  <hh.sequence>
	    <hh.pause/>
	    <hh.emit S1/>
	    <hh.exit T1/>
	  </hh.sequence>
	  <hh.loop>
	    <hh.trap T2>
	      <hh.local S2>
		<hh.parallel>
		  <hh.sequence>
		    <hh.pause/>
		    <hh.emit S2/>
		    <hh.exit T2/>
		  </hh.sequence>
		  <hh.loop>
		    <hh.sequence>
		      <hh.if S1>
			<hh.if S2>
			  <hh.emit S1_and_S2/>
			  <hh.emit S1_and_not_S2/>
			</hh.if>
			<hh.if S2>
			  <hh.emit not_S1_and_S2/>
			  <hh.emit not_S1_and_not_S2/>
			</hh.if>
		      </hh.if>
		      <hh.pause/>
		    </hh.sequence>
		  </hh.loop>
		</hh.parallel>
	      </hh.local>
	    </hh.trap>
	  </hh.loop>
	</hh.parallel>
      </hh.local>
    </hh.trap>
  </hh.loop>
</hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "P18");
