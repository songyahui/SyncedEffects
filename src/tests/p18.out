P18> ;
--- Output: not_S1_and_not_S2
P18> ;
--- Output: S1_and_S2 S1_and_not_S2 not_S1_and_not_S2
P18> ;
--- Output: S1_and_S2 S1_and_not_S2 not_S1_and_not_S2
P18> ;
--- Output: S1_and_S2 S1_and_not_S2 not_S1_and_not_S2
